mlflow/utils/environment.py

```diff
@@ -82,3 +89,61 @@ def _get_additional_pip_dep(conda_env):
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
     pip_deps = _get_additional_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _get_pip_requirements(module_names):
+    module_to_packages = importlib_metadata.packages_distributions()
```

Comment:
Does `packages_distributions` also give us transitive dependency versions? If not, can we extract that information using other APIs from `importlib_metadata`?